modules/godot/systems/timer_updater_system.cpp

```diff
@@ -0,0 +1,30 @@
+#include "timer_updater_system.h"
+#include "core/os/os.h"
+#include "scene/main/scene_tree.h"
+
+void timer_updater_system(TimersDatabag *td,
+		SceneTreeInfoDatabag *p_sti,
+		OsDatabag *p_os) {
+	ERR_FAIL_COND_MSG(td == nullptr, "The `TimersDatabag` `Databag` is not part of this world. Add it please.");
+	ERR_FAIL_COND_MSG(p_sti == nullptr, "The `SceneTreeDatabag` `Databag` is not part of this world. Add it please.");
+	ERR_FAIL_COND_MSG(p_os == nullptr, "The `OsDatabag` `Databag` is not part of this world. Add it please.");
+	ERR_FAIL_COND_MSG(p_os->get_os() == nullptr, "No OS found. This should not happen.");
+
+	const bool paused = p_sti->is_paused();
+	const uint64_t old_now = td->internal_get_full_now();
+	const uint64_t old_pause = td->internal_get_pause();
+	const uint64_t new_now = p_os->get_os()->get_ticks_usec();
+
+	td->internal_set_now(new_now);
+	if (paused) {
+		if (old_pause < new_now) {
+			td->internal_set_pause(old_pause + (new_now >= old_now ? new_now - old_now : old_now - new_now)); // Why are you going back?
```

Comment:
I think we should just thrown an error here instead, since this can't really happen, or at least it's not supposed to happen right?

Reply:
Yeah, true. I kept it because I didn't know if negative delta time was legal. Removed now.